gcce.service.js

```diff
@@ -378,30 +378,33 @@ module.exports = class GoogleComputeService extends Compute {
     return (await iam.projects.serviceAccounts.list(request)).data.accounts;
   }
 
-  async listNetworks(_, fields, pageToken) {
+  async listNetworks({ query }, fields, pageToken) {
     const request = removeUndefinedAndEmpty({
       auth: this.getAuthClient(),
       project: this.projectId,
       fields: parseFields(fields),
+      filter: query ? `name:${query}` : undefined,
```

Comment:
If the `query` is falsy, the object will have a property `filter` with `undefined` value. Such assignment will not make the property disappear from the object.
The only (non-ninja) way to achieve that is to conditionally assign a value after an object is defined.

Reply:
That's why this object gets passed to the `removeUndefinedAndEmpty` function